src/lib.rs

```diff
@@ -2992,10 +2995,7 @@ pub unsafe trait AsBytes {
         Self: NoCell,
     {
         let start = bytes.len().checked_sub(mem::size_of_val(self))?;
-        bytes
-            .get_mut(start..)
-            .expect("`start` should be in-bounds of `bytes`")
-            .copy_from_slice(self.as_bytes());
+        bytes.get_mut(start..)?.copy_from_slice(self.as_bytes());
```

Comment:
Can you add a comment here explaining why we're using `?` rather than unwrapping? IIUC, the idea is that we know this will always succeed, so in principle unwrapping should be fine, but we don't want to risk the compiler not being able to optimize out the panic in all cases. Is that right?

Also, can you leave a TODO comment linking to #202 to improve this eventually? See other TODO comments for the syntax.

Reply:
> Also, can you leave a TODO comment linking to #202 to improve this eventually? See other TODO comments for the syntax.

I don't understand what you're looking for here. Is there a further improvement that should be made to `write_to_suffix`'s implementation? Or are you referring to better checking (e.g. redpen)?

Reply:
Yeah, I'd like to be able to go back to panicking once we can prove that the panic is optimized out, or ideally to some other proven-with-the-type-system approach that gives us a guarantee that there will be no panic. The problem with `?` is that it won't panic, but might introduce a new branch, and that's harder for us to detect. By contrast, if we can detect panics with something like redpen, then we can have higher confidence that the method is well-optimized.

Reply:
Okay, after https://github.com/google/zerocopy/issues/202#issuecomment-1979344700 I added a comment indicating that we are using `?` rather than `unwrap`/`expect` because it is lighter-weight.